### PR TITLE
Improved CLI, templates, output section, first test

### PR DESCRIPTION
**Pull request by anton-khodak**
- changed the structure of templates so that they can be divided into smaller parts
- rewrote interface to `argparse2cwl`, now a user can get information about `a2c` arguments with passing `-ha` or `--help_arg2cwl` option to any tool he/she wants to wrap.
- added an option of plugging hand-written output section to CWL tool
- removed `posititon` field for optional arguments
- simple general test

### CODE REVIEW

File: argparse/__init__.py [hunk 1]
```diff
@@ -65,9 +65,26 @@ def load_conflicting_package(name, not_name, local_module):
 tools = []
 
 
+def get_arg2cwl_parser():
+    help_text = """
+    argparse2cwl forms CWL command line tool from Python tool
+    """
+    arg2cwl_parser = ap.ArgumentParser(prog=sys.argv[0], description=help_text, add_help=False)
+    arg2cwl_parser.add_argument('tools', nargs='*',
+                                help='Command for running your tool(s), without arguments')
```
Comment (mr-c) [inline argparse/__init__.py:74]: Is this a required option or does it override the default?

I would give an example: `python -m modulename` or `my_script`

File: argparse/__init__.py [hunk 2]
```diff
@@ -65,9 +65,26 @@ def load_conflicting_package(name, not_name, local_module):
...
+    arg2cwl_parser.add_argument('-d', '--directory',
+                                help='Directory for placing formed CWL tool')
```
Comment (mr-c) [inline argparse/__init__.py:76]: I would word this as

> Directory to store CWL tool descriptions

File: argparse/__init__.py [hunk 3]
```diff
@@ -65,9 +65,26 @@ def load_conflicting_package(name, not_name, local_module):
...
+    arg2cwl_parser.add_argument('-b', '--basecommand',
+                                help='Command that appears in `basecommand` field in CWL tool')
```
Comment (mr-c) [inline argparse/__init__.py:78]: Again, is this a required option or does it override a normally correct default?

File: argparse/__init__.py [hunk 4]
```diff
@@ -65,9 +65,26 @@ def load_conflicting_package(name, not_name, local_module):
...
+    arg2cwl_parser.add_argument('-o', '--output_section',
+                                help='File with output section which will be appended to a formed CWL tool')
```
Comment (mr-c) [inline argparse/__init__.py:80]: It isn't just appended, it replaces the autogenerated `outputs` section.

File: argparse/__init__.py (hunk 1 above)
Comment (anton-khodak) [inline argparse/__init__.py:74]: Could you please explain your question in more detail?

File: argparse/__init__.py (hunk 3 above)
Comment (anton-khodak) [inline argparse/__init__.py:78]: It overrides a normally correct default. This option is intended to cover shebang cases, when, for instance, a program is running like `./program.py`, but a developer wants to have `basecommand: python` and `program.py` as an input in a CWL tool.

File: argparse/__init__.py (hunk 2 above)
Comment (anton-khodak) [inline argparse/__init__.py:76]: I took your comments into account in the following PR

File: argparse/__init__.py (hunk 3 above)
Comment (mr-c) [inline argparse/__init__.py:78]: Great; then the description can be modified to make that more explicit.

**Pull request merged**